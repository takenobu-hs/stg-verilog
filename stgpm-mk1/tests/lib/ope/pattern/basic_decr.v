

`define SP1_DECR

  parameter DW = 32;


  /* test pattern */

  initial begin
    @(negedge rst);
    repeat(2) @(posedge clk);

    @(posedge clk);
    a = 'd2;

    @(posedge clk);
    a = 32'h1234_5678;

    @(posedge clk);
    a = 32'h0000_0001;

    /* carry */
    @(posedge clk);
    a = 32'h0000_0000;

    repeat(5) @(posedge clk);
    halt();
  end


