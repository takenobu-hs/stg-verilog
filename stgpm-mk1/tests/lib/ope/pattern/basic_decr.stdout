monitor   :       0: xxxxxxxx  x xxxxxxxx
monitor   :      80: 00000002  0 00000001
monitor   :      90: 12345678  0 12345677
monitor   :     100: 00000001  0 00000000
monitor   :     110: 00000000  1 ffffffff
halt      : halt
